backend/src/test/java/de/bund/digitalservice/ris/search/integration/service/SearchTermServiceTest.java

```diff
@@ -0,0 +1,114 @@
+package de.bund.digitalservice.ris.search.integration.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import de.bund.digitalservice.ris.search.models.ParsedSearchTerm;
+import de.bund.digitalservice.ris.search.service.SearchTermService;
+import java.util.List;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+@Tag("integration")
+class SearchTermServiceTest {
+
+  @Autowired SearchTermService searchTermService;
+
+  @Test
+  void testEmptyString() {
+    ParsedSearchTerm result = searchTermService.parse("");
+
+    assertTrue(result.unquotedTokens().isEmpty());
+    assertTrue(result.quotedSearchPhrases().isEmpty());
+  }
+
+  @Test
+  void testOnlyUnquotedTerms() {
+    ParsedSearchTerm result = searchTermService.parse("hello world");
+
+    assertEquals(List.of("hello", "world"), result.unquotedTokens());
+    assertTrue(result.quotedSearchPhrases().isEmpty());
+  }
+
+  @Test
+  void testOnlyQuotedTerms() {
+    ParsedSearchTerm result = searchTermService.parse("\"hello world\"");
+    assertTrue(result.unquotedTokens().isEmpty());
```

Comment:
More of a note (maybe worth a topic in the dev round), I'd prefer the more verbose/readable `assertThat(...)...` syntax from the `AssertJ` liberary. 
E.g.
`assertThat(result.unquotedTokens()).isEmpty();`
or 
`assertThat(result.quotedSearchPhrases()).containsExactly("hello world")`
(I know these are just the tests from the former QuotedStringParser and they were already like that)